File: dev/rabbitmq_python/send.rb
#!/usr/bin/env ruby
# encoding: utf-8

require "bunny"

conn = Bunny.new(:host=>"172.17.0.5", :port=>5672)
conn.start

ch = conn.create_channel
x = ch.topic("david")
routing_key = "hello"
msg = "Hello World!"

x.publish(msg, :routing_key => routing_key)
puts " [x] Sent '#{msg}'"

conn.close

File: dev/rabbitmq_python/send.py
#!/usr/bin/env python
import pika
import sys

connection = pika.BlockingConnection(pika.ConnectionParameters(host='172.17.0.5'))
channel = connection.channel()

channel.exchange_declare(exchange='david',
                         type='topic')

routing_key = 'hello'

message = 'Hello World!'
channel.basic_publish(exchange='david',
                      routing_key='hello',
                      body=message)
print(" [x] Sent %r" % (message))
connection.close()
